### PR TITLE
Added perl environment

### PR DESCRIPTION
**Pull request by LittleFox94**
New environment for fission: Perl.

This is currently just a prototype, but may already be useful for some
tasks.

Refs #219

### CODE REVIEW

File: environments/perl/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Fission Perl environment
+
+This environment is based on Dancer2 and Twiggy.
+
+You can use AnyEvent here.
```
Comment (erwinvaneyk) [inline environments/perl/README.md]: I am not familiar with Perl, so AnyEvent, Dancer2 or Twiggy do not say much. Could you maybe make this a bit more descriptive and/or add links?

File: environments/perl/server.pl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    post '/specialize' => sub {
+        if($userfunc) {
+            send_error('Already done', 500);
+        }
+
+        $userfunc = require($options->{codepath});
+        return '';
+    };
+
+    get '/' => sub {
```
Comment (erwinvaneyk) [inline environments/perl/server.pl]: Is it possible to make this endpoint respond to any http method? Such that you could use GET, POST, DELETE etc. to make it consistent with other environments.

File: environments/perl/server.pl [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+{
+    package App::Fission::Perl;
+
+    use Dancer2;
+
+    our $userfunc;
+
+    post '/specialize' => sub {
+        if($userfunc) {
+            send_error('Already done', 500);
```
Comment (erwinvaneyk) [inline environments/perl/server.pl]: `send_error('Not a generic container', 400);`

File: environments/perl/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+## Example
+
+```perl
+# hello.pm
+return sub {
+    return 'Hello, Perl!';
+}
```
Comment (erwinvaneyk) [inline environments/perl/README.md]: Could you make this a (also) a separate file. That way users can add the function easily using `fission fn create --name foo --code https://github.com/fission/fission/blob/master/environments/perl/hello.pm` to try it out.

Comment (LittleFox94) [reply environments/perl/README.md]: Added hello.pm for this, this one got more documentation.

**Comment by erwinvaneyk** [comment]
lgtm 👍  @soamvasani

**Comment by soamvasani** [comment]
Awesome to see a Perl environment!  Thanks @LittleFox94 !

**Comment by soamvasani** [comment]
@LittleFox94 it would also be useful to add a few more example functions, like how to access request params, how to log, etc.

**Comment by LittleFox94** [comment]
@soamvasani ok, I'll add some more examples in my lunch break :)

**Pull request merged**